ops/model.py

```diff
@@ -954,7 +954,7 @@ def fetch(self, name: str) -> Path:
         on disk, otherwise it raises a ModelError.
         """
         if name not in self._paths:
-            raise RuntimeError('invalid resource name: {}'.format(name))
+            raise ModelError('invalid resource name: {}'.format(name))
```

Comment:
So this isn't about it being found on disk, but about being found in the metadata definition for the charm.

The doc string is talking about "If you ask for a resource and juju cannot give that to you, it is a ModelError". I'm open to the discussion of "I wrote code that asked for X but never defined what X is".

This is logically closer to a `NameError` (eg, if you call foob() but the function is foo() you get a NameError in python)
I do think ModelError is something you might catch, while *this* is indicative of a coding error and probably shouldn't be caught.
What do you think about NameError or KeyError?
KeyError follows a model of dicts where it seems you might add the key, which I don't think we want here.

Reply:
Yeh, I think `NameError` is probably appropriate. I'll update this PR.

Reply:
Agree that NameError is appropriate here.